### ISSUE TITLE
LNK File created on Windows 2008 Server Fails

### ISSUE THREAD

**Issue by einsty** [codebling/WindowsShortcuts#4]
Thanks to _everyone_ behind this little gem of code.   We've found it useful in a recent project task.

however, we did run into an odd situation whereby a LNK file created on a Windows 2008 server failed... but oddly, a lnk file created on a Windows XP machine and links created on Windows 10 seem to work fine.

When parsing the problematic link file, we saw the fail occur in the following block within the `has_working_directory` flag. ``` final int string_len = bytesToWord(link, next_string_start) * 2; // times 2 because UTF-16```

If this rings a bell for anybody, give a shout.

**Comment by codebling** [comment]
What exception are you seeing?

**Comment by codebling** [comment]
I don't have a Windows 2008 Server instance hanging around anywhere that I can try this on, so any additional info that you can give would be great. You mention a failure, what exactly is the nature of that failure?  Is it an exception? If yes, please provide the exception with stack trace. If not, is it simply a failure of the code to parse that specific link? In what way does it fail, and is it on that specific line?  

You mention a problematic link file, are you able to attach it?

**Comment by einsty** [comment]
Hello and sure thing!  The error was `java.lang.ArrayIndexOutOfBoundsException: 3015410` and  then the `java.text.ParseException: Could not be parsed, probably not a valid WindowsShortcut` kicks in.

The files attached actually error in the `has_relative_path` evaluation, not the `has working directory` as i stated above, I may have mis-stated originally.  

I am attaching a screen shot of the values in IntelliJ debugger from right before the exception, and also the actual file that is failing for us.

Thank you greatly for even taking a look!
<img width="328" alt="skitch (14)" src="https://user-images.githubusercontent.com/1541350/56865385-e0d97480-6992-11e9-8ce2-cff99a9d466b.png">

[Windows 2008 Server Links.zip](https://github.com/codebling/WindowsShortcuts/files/3125042/Windows.2008.Server.Links.zip)

**Comment by codebling** [comment]
Just a quick update to say that I've started looking at this but haven't found anything yet. Will check again

**Comment by codebling** [comment]
@einsty have you made any progress on this? It's been an incredibly busy week for me and I have not done any debugging yet.

**Comment by codebling** [comment]
I'm looking at this once again but I'm having trouble diagnosing the issue. Have you made any progress on this? 

There is also a project here which seems to be pretty good and might fix the issue you're having
https://github.com/BlackOverlord666/mslinks

**Comment by einsty** [comment]
Thanks for the continued effort on this @codebling .   Our client has just avoided the creation of link files on the Windows Server 2008 platform and so far this is all working out OK.  So, although we never got to the bottom of it code-wise, it is resolved operationally. 

Given the above, and the link to the other project, I would be OK if the issue is closed.

**Comment by codebling** [comment]
I added the files to the [repo as test fixtures](https://github.com/codebling/WindowsShortcuts/tree/master/src/test/fixtures), I hope this is ok!  

With regards to closing, maybe we can leave the issue open so that it's visible to other potential users? I'm not sure if, as the author of an isssue, one can unsubscribe, but if you don't wish to receive updates I can close this and open a different issue. Maybe it makes more sense to do that anyways... ok created a new issue. Closing this!
